### PR TITLE
Add control plane upgrades

### PR DESCRIPTION
**Pull request by janboll**
Redoing https://github.com/app-sre/qontract-reconcile/pull/3494 with bugfixes:

- [3237d81](https://github.com/app-sre/qontract-reconcile/pull/3500/commits/3237d816211f08cca27417d30f12415d4400e089)
- [4caf462](https://github.com/app-sre/qontract-reconcile/pull/3500/commits/4caf46291948583f0b78a1071829eb96ee24f901)

https://issues.redhat.com/browse/APPSRE-7243

### CODE REVIEW

File: reconcile/openshift_upgrade_watcher.py [hunk 1]
```diff
@@ -127,9 +127,7 @@ def run(
     if defer:
         defer(state.cleanup)
 
-    clusters = [
-        c for c in get_clusters() if c.ocm and not (c.spec and c.spec.hypershift)
-    ]
+    clusters = [c for c in get_clusters() if c.ocm]
```
Comment (kwilczynski) [inline reconcile/openshift_upgrade_watcher.py:130]: Would Set work better over a list here?

**Review by kwilczynski — APPROVED** [review]
@janboll, looks good!

That said, it could use another pair of eyes, especially from someone who knows our upgrades a bit better than I do.

**Review by patjlm — APPROVED** [review]
the 2 commits look good to me, thanks!

**Pull request merged**